### ISSUE TITLE
Suddenly "Your URL didn't respond with the value of the challenge parameter."

### ISSUE THREAD

**Issue by mEzzIbr** [slackapi/bolt-python#927]
Hi there,

My bot was working very well, for almost one year.

python script with Flask and Slack bolt frameworks

[slack-sdk<4,>=3.17.2 in venv\lib\site-packages (from slack_bolt) (3.18.3)]

but suddenly when I run the script locally and host it in Ngrok then copy the generated ngrok URL in "New Request URL" Input inside Slack events subscriptions page, it always shows me an Error Message whose title "Your request URL responded with an HTTP error. Update your URL to receive a new request and challenge value." - "Your URL didn't respond with the value of the challenge parameter."

I didn't change anything at all! I tried many solutions I pull an old commit with an old version of the script and it didn't work the same problem was shown

I've already checked the signing secret and recopy it, and nothing happened, I Regenerated it and copied the new one, and nothing happened, I've also checked the bot token 

Everything as it is, I don't know what's the problem 

So, I NEED your help, please

Thanks in Advance

![image](https://github.com/slackapi/bolt-python/assets/59320267/86cc8246-e082-421e-ae7a-c5f1639516e1)
![image](https://github.com/slackapi/bolt-python/assets/59320267/7c145ae5-f20a-41ab-88a0-dbbef124f101)
![image](https://github.com/slackapi/bolt-python/assets/59320267/98fc041d-e07c-4897-9ba5-96227383978f)

**Comment by seratch** [comment]
Hi @mEzzIbr, thanks for asking the question!

According to the screenshot that you've shared, it seems that you were trying to set `/slack/events/` instead of `/slack/events`. Since your Flask app does not serve the path ending with `/`, perhaps your app responded with 404 Not Found for the URL verification request. You can use `/slack/events` instead and it should work for you.

**Comment by mEzzIbr** [comment]
Hi @seratch, Thank you so much for your response.

Actually, I just put "/" before the screenshot just to try any change to refresh the error message,

In the normal tries, I don't put it, Now as it's shown on the screenshot, there is no "/", and the error is still here.

So I don't think that is the problem.

I'd be appreciated again if there are any suggestions about my problem.

Thank you

![image](https://github.com/slackapi/bolt-python/assets/59320267/7d02a417-fb7c-4bbe-9f8b-9426f040248e)

**Comment by seratch** [comment]
I recommend checking `http://localhost:4040/`, which provides more information about the received requests and how your app responded. If you don't see any requests there, the URL you're trying to set may be wrong. If your app respods with non-200 status (or it might not respond at all), your app configuration and/or listening port should be incorrect.

**Comment by seratch** [comment]
I just noticed that you're trying to connect the ngrok endpoint to `localhost:8080` (according to the screenshot you've shared). Is it intentional? If your local Python app listens on a different port like 3000, you need to start the ngrok process like `ngrok http 3000`.

**Comment by mEzzIbr** [comment]
I rerun it on 4040 port, then change the Ngork URL and put it on Slack, Sadly the same problem

![image](https://github.com/slackapi/bolt-python/assets/59320267/c2eeacc4-7d18-43bc-97fc-d747e014f2a8)
![image](https://github.com/slackapi/bolt-python/assets/59320267/aa1e0809-fea4-4b5a-8085-8b11784c1170)

**Comment by seratch** [comment]
4040 is used by ngrok. Please go with another port.

**Comment by mEzzIbr** [comment]
I tried also 3000

![image](https://github.com/slackapi/bolt-python/assets/59320267/17018c40-3b08-48e0-80f6-a21d4381a4a7)
![image](https://github.com/slackapi/bolt-python/assets/59320267/8ab28b31-96c4-4191-8e2e-c9565f1b8810)

**Comment by seratch** [comment]
What happens when you access the ngrok URL in your web browser? Do you see any request logs on the ngrok process terminal console? If not, something wrong may be happening with your ngrok usage or your network settings.

If you see requests through the ngrok URL but your local app process fails to handle it, how about checking if localhost:3000 responds to your web browser request first? Your local app and/or listening port may have some issues.

I recommend narrowing the problem down this way.

**Comment by mEzzIbr** [comment]
Currently, I'm trying google cloud (cloud run service).

The error changed a little as it's shown in the screenshot, and I got that log which may help.

![image](https://github.com/slackapi/bolt-python/assets/59320267/ea3565c3-f258-465a-8f6f-6a226b11429e)
![image](https://github.com/slackapi/bolt-python/assets/59320267/c8f04951-3b47-4bfc-a597-408a37e82542)

**Comment by mEzzIbr** [comment]
Hi @seratch ,

It was ngrok! 

when I logged in with an account (not anonymous) and started to run ngrok through a token or key; it finally worked Well!

It seems that ngrok updated some requirements and specs or something like that in their terminal, I hope my issue finds its road to anyone facing the same problem to know the cause of it quickly instead of spending many days in futile attempts (Like I did)

Thank you.
